src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.cs

```diff
@@ -1,52 +1,186 @@
+    ///
+    /// If a target assembly needs additional AssemblyResolve to satisfy its dependencies of for any other reasons,
+    /// it should set up its own as the first thing after its entry point is called.
+    ///
+    /// ! Do not make the AppDomain.AssemblyResolve handler in here more complex !
+    /// If anything, it should be simplified and any special logic should be moved into the respective assemblies
+    /// requested for loading.
+    ///
+    /// ! Also, remember that this assembly is shared between the Tracer's profiler component
+    /// and the Profiler's profiler component. DO not put specialized code here !
     /// </summary>
     public partial class Startup
     {
         /// <summary>
-        /// Initializes static members of the <see cref="Startup"/> class.
-        /// This method also attempts to load the Datadog.Trace.ClrProfiler.Managed .NET assembly.
+        /// The constants <c>TargetLibraryEntrypointMethod</c>, <c>...Type</c> and <c>...Namespace</c> specify
+        /// which entrypoint to call in the specified assemblies. The respective assembly is expected to have
+        /// exactly this entry point or nothing will be invoked.
+        /// The method must be static, the return type of the method must be <c>void</c> and it must have no parameters.
+        /// Before doing anything else, the target assemblies must set up AppDomain AssemblyResolve events that
+        /// make sure that their respective dependencies can be loaded.
         /// </summary>
-        static Startup()
+        public const string TargetLibraryEntrypointMethod = "Run";
+
+        /// <summary> <see cref="Startup.TargetLibraryEntrypointMethod" /> </summary>
+        public const string TargetLibraryEntrypointType = "DllMail";
+
+        /// <summary> <see cref="Startup.TargetLibraryEntrypointMethod" /> </summary>
+        public const string TargetLibraryEntrypointNamespace = "Datadog.AutoInstrumentation";
+
+        private const string TargetLibraryEntrypointFullTypeName = TargetLibraryEntrypointNamespace
+                                                                 + "."
+                                                                 + TargetLibraryEntrypointType;
+
+#pragma warning disable SA1308 // Variable names must not be prefixed (if not this, what is the static prefix?)
+        private static string s_managedProfilerDirectory = null;
+#pragma warning restore SA1308 // Variable names must not be prefixed
+
+        private readonly string[] _assemblyNames;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Startup"/> class.
+        /// </summary>
+        /// <param name="assemblyNames">List of assemblies to load ans start.</param>
+        public Startup(string[] assemblyNames)
+        {
+            _assemblyNames = assemblyNames;
+        }
+
+        /// <summary>
+        /// Instntites a <c>Startup</c> instance with the specified assemblies and executes it.
```

Comment:
```suggestion
        /// Instantiates a <c>Startup</c> instance with the specified assemblies and executes it.
```

Reply:
Updated in _macrogreg/loader-with-modules-cctor-and-managed-loader_-branch.